lib/common_ui/primary_button.dart

```diff
@@ -23,6 +25,9 @@ class PrimaryButton extends StatelessWidget {
 
   onPressedButton(BuildContext context) {
     SchedulerBinding.instance!.addPostFrameCallback((_) {
+      StudentsLogic provider = context.read<StudentsLogic>();
```

Comment:
Why is there a separate class for "Logic". The "Logic" should be inside the "Student" class. That is what classes are for.

Also, it is for one "Student" and not "Students". Singular vs plural can make reading code much more complicated.